tests/baselines/reference/parameterPropertyInConstructor4.js

```diff
@@ -0,0 +1,28 @@
+//// [tests/cases/compiler/parameterPropertyInConstructor4.ts] ////
+
+//// [parameterPropertyInConstructor4.ts]
+export class C {
+    constructor(public a: number[] = [], b: number) {
+    }
+}
+
+
+//// [parameterPropertyInConstructor4.js]
+"use strict";
+Object.defineProperty(exports, "__esModule", { value: true });
+exports.C = void 0;
+var C = /** @class */ (function () {
+    function C(a, b) {
+        if (a === void 0) { a = []; }
+        this.a = a;
+    }
+    return C;
+}());
+exports.C = C;
+
+
+//// [parameterPropertyInConstructor4.d.ts]
+export declare class C {
+    a: number[] | undefined;

```

Comment:
This is incorrect
```suggestion
    a: number[];

```
since the `= []` ensures that the property is always assigned, so only the parameter type should be `| undefined`.

Reply:
@weswigham thanks for your review. I thought that both parameter and property declarations should use the same type...

Reply:
@weswigham could you review the latest changes? thanks

Reply:
@weswigham could you review the latest changes? thanks

Reply:
@weswigham could you review the latest changes? thanks

Reply:
@weswigham could you review the latest changes? thanks

Reply:
Yeah, you can see the difference in an example like
```ts
class C {
	constructor(public arr: number[] = [], t: number) {}
}

const a = new C(undefined, 0); // allowed

a.arr = undefined; // is an error
```
The parameter and the property actually do have different internal types. The printer *might* not always be getting that correct right now with some recent changes to how nodes are reused, but this is definitely a step in the right direction.